onnxruntime/core/providers/cuda/math/matmul.cc

```diff
@@ -180,6 +180,11 @@ Status MatMul<T>::ComputeInternal(OpKernelContext* ctx) const {
   ORT_RETURN_IF_ERROR(right_arrays.CopyToGpu(ctx->GetComputeStream()));
   ORT_RETURN_IF_ERROR(output_arrays.CopyToGpu(ctx->GetComputeStream()));
 
+#if defined(CUDA_VERSION) && CUDA_VERSION >= 11000
```

Comment:
1. We no longer support CUDA 10.x, so this "#if" is not needed. The minimum CUDA version we support is CUDA 11.4. 
2. When there is a need to use CUDA_VERSION, we should use cudaRuntimeGetVersion() instead. They might be different. The later one is more accurate.